cmd/dockerd/daemon.go

```diff
@@ -172,14 +171,12 @@ func (cli *DaemonCli) start(opts *daemonOptions) (err error) {
 	}
 
 	ctx, cancel := context.WithCancel(context.Background())
-	waitForContainerDShutdown, err := cli.initContainerD(ctx)
-	if waitForContainerDShutdown != nil {
-		defer waitForContainerDShutdown(10 * time.Second)
-	}
+	waitForContainerdShutdown, err := cli.initContainerd(ctx)
 	if err != nil {
 		cancel()
 		return err
 	}
+	defer waitForContainerdShutdown()
 	defer cancel()
```

Comment:
Pet peeve:
```suggestion
	defer cancel()
	waitForContainerdShutdown, err := cli.initContainerd(ctx)
	if err != nil {
		return err
	}
	defer waitForContainerdShutdown()
```

Reply:
Ah, yes, I was looking at that one, and wasn't sure if the order in which they're called was important 🤔

Reply:
done